pkg/systemd/host.js

```diff
@@ -663,9 +664,44 @@ PageSystemInformationChangeSystime.prototype = {
         $('#systime-timezones').on('change', enable_apply_button);
         $('#systime-date-input').on('focusin', $.proxy(this, "store_date"));
         $('#systime-date-input').on('focusout', $.proxy(this, "restore_date"));
+
+        this.ntp_servers_tmpl = $("#ntp-servers-tmpl").html();
+        Mustache.parse(this.ntp_servers_tmpl);
+
+        this.ntp_status_tmpl = $("#ntp-status-tmpl").html();
+        Mustache.parse(this.ntp_status_tmpl);
+
+        $('#systime-ntp-servers').on('click', '[data-action="add"]', function () {
+            var index = $(this).attr('data-index');
+            if (self.ntp_servers !== undefined) {
+                self.sync_ntp_servers();
+                self.ntp_servers.splice(index+1, 0, "");
+                self.update_ntp_servers();
+            }
+            // HACK - without returning 'false' here, the dialog will
+            // be magically closed when controlled by the
+            // check-system-info test.
```

Comment:
Is this because you use 'data-action="add"'? What happens when you change it to 'data-actionx'?